### PR TITLE
histogram: allow to reset histogram measurement.

### PR DESCRIPTION
**Pull request by shelly-lb**
This allows to reset histogram information, which is usefull when investigating scenario.
Reset is working by writing 1 to "reset" file that is provided with the dataset.

Signed-off-by: Shelly Kagan <shelly@lightbitslabs.com>

### CODE REVIEW

File: src/procstat.c [hunk 1]
```diff
@@ -1203,6 +1203,17 @@ static ssize_t procstat_fmt_u32_percentile(void *object, uint64_t arg, char *buf
...
 void procstat_histogram_u32_add_point(struct procstat_histogram_u32 *series, uint32_t value)
 {
+	unsigned reset;
+
+	reset = __atomic_load_n(&series->reset, __ATOMIC_RELAXED);
+	if (reset) {
+		series->count = 0;
+		series->sum = 0;
+		series->last = 0;
+		memset(series->histogram, 0, PROCSTAT_PERCENTILE_ARR_NR * sizeof *series->histogram);
```
Comment (sashas-lb) [inline src/procstat.c]: i think there are backets missing

File: src/procstat.c [hunk 2]
```diff
@@ -1281,7 +1307,12 @@ int procstat_create_histogram_u32_series(struct procstat_context *context, struc
 		file->arg = i;
 	}
 
+	control = create_control_file(context, &series_stat->root, "reset", series_stat, reset_histogram_u32_series);
+	if (!control)
+		goto fail_remove_stat;
+
 	return 0;
+
```
Comment (sashas-lb) [inline src/procstat.c]: redundant space

File: src/procstat.h [hunk 3]
```diff
@@ -275,13 +275,14 @@ typedef void (*percentiles_calculator)(uint32_t *histogram,
...
 #define MAX_SUPPORTED_PERCENTILE 20
 struct procstat_histogram_u32 {
-	uint64_t 			  sum;
-	uint64_t 			  count;
-	uint64_t 			  last;
-	int	 			  npercentile;
-	struct procstat_percentile_result percentile[MAX_SUPPORTED_PERCENTILE];
-	uint32_t 			  *histogram;
-	percentiles_calculator 		  compute_cb;
+	uint64_t				sum;
```
Comment (sashas-lb) [inline src/procstat.h]: please split to another  commit

**Comment by sashas-lb** [comment]
nice JOB !!! 
note that we have to do the follwoing:
1) solve race condition for reset
2) return zeroed values between reset and "host path" 
3) poor mans test :) like in series

**Comment by shelly-lb** [comment]
@sashas-lb DONE :) review and merge if ok

File: src/procstat.c [hunk 4]
```diff
@@ -1195,42 +1195,113 @@ void procstat_loop(struct procstat_context *context)
...
 {
+	unsigned reset;
+
+	reset = __atomic_load_n(&series->reset, __ATOMIC_RELAXED);
+	if (reset) {
+		series->count = 0;
+		series->sum = 0;
+		series->last = 0;
+		memset(series->histogram, 0, PROCSTAT_PERCENTILE_ARR_NR * sizeof(*series->histogram));
+
```
Comment (sashas-lb) [inline src/procstat.c]: please remove space

File: src/procstat.c [hunk 5]
```diff
@@ -1195,42 +1195,113 @@ void procstat_loop(struct procstat_context *context)
...
+		goto write_var;
+	default:
+		return -1;
+	}
+write_nan:
+	return snprintf(buffer, len, "nan");
+write_var:
+	return procstat_format_u64_decimal(data_ptr, arg, buffer, len);
+}
+
```
Comment (sashas-lb) [inline src/procstat.c]: please remove redundant space

File: src/procstat.c (hunk 4 above)
Comment (shelly-lb) [reply src/procstat.c]: done

File: src/procstat.c (hunk 5 above)
Comment (shelly-lb) [reply src/procstat.c]: done

**Pull request merged**